//Промисы: сравните then и catch.
// Являются ли фрагменты кода ниже эквивалентными? Другими словами, ведут ли они себя одинаково во всех обстоятельствах, для всех переданных им обработчиков?
//
// promise.then(f1).catch(f2);
// Против:
//
// promise.then(f1, f2);

// Нет, не являются.
// .then передаёт результат или ошибку следующему блоку. Так как в первом примере в цепочке далее имеется блок catch, а во втором – нет, то ошибка в нём останется необработанной.